### PR TITLE
Login and link capabilities

### PR DESCRIPTION
**Pull request by pagbrl**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,72 +1,80 @@
...
+            "git add",
+            "eslint --fix",
+            "git add"
+        ],
+        "*.{json,md}": [
+            "prettier --write",
+            "git add"
+        ]
+    },
+    "zola": {
```
Comment (pagbrl) [inline package.json:75]: Our own package.json field 😄

File: index.js [hunk 2]
```diff
@@ -1,12 +1,120 @@
...
 const getRandom = (min, max) => Math.floor(Math.random() * max) + min
+const zolaApiUrl = 'https://api.zola.ink/graphql'
+const zolaAppUrl = 'https://app.zola.ink'
 
 program.version('0.2.0')
 
+// We need this because fetch has an issue with our SSL certs atm
+process.env['NODE_TLS_REJECT_UNAUTHORIZED'] = 0
+
+const retrieveLogin = async () => {
```
Comment (pagbrl) [inline index.js]: Good thing is we can use this anywhere like this 
```
const { user, token } = await retrieveLogin()
```
To secure the command (require login). The retrieveLogin method will throw an error and exit if the user has no configuration.

Comment (Tom-Bonnike) [reply index.js]: W0W

File: index.js [hunk 3]
```diff
@@ -1,12 +1,120 @@
...
+          'You are not logged in, please use the login command before running this command.\n'
+        )
+      )
+
+      // We write empty values here
+      await fs.writeJson(path.join(configDir, 'config.json'), {
+        email: '',
+        token: ''
+      })
+      process.exit(1)
```
Comment (Tom-Bonnike) [inline index.js]: should we exit with an error?

File: index.js [hunk 4]
```diff
@@ -1,12 +1,120 @@
...
+  fetch.use(({ request, options }, next) => {
+    if (!options.headers) {
+      options.headers = {} // Create the headers object if needed.
+    }
+
+    options.headers['authorization'] = `Bearer ${token}`
+    next()
+  })
+
+  fetch({
```
Comment (Tom-Bonnike) [inline index.js]: lets switch to async await pls

File: index.js [hunk 5]
```diff
@@ -1,12 +1,120 @@
...
+  }).then(async res => {
+    const data = res.data.getProject
+
+    if (data === null) {
+      res.errors.forEach(error => {
+        console.log(chalk.bgRed.white(error.message))
+      })
+      console.log('\n')
+      const json = await fs.readJson(path.join('./', 'package.json'))
+      delete json.zola
```
Comment (Tom-Bonnike) [inline index.js]: lol

File: package.json (hunk 1 above)
Comment (Tom-Bonnike) [reply package.json:75]: WOW!

**Review by Tom-Bonnike — APPROVED** [review]

File: src/login.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
+const path = require('path')
+const homeDir = require('os').homedir()
+const fs = require('fs-extra')
+const chalk = require('chalk')
+const configDir = `${homeDir}/.config/zola`
+const { createApolloFetch } = require('apollo-fetch')
+const inquirer = require('inquirer')
+
+const zolaApiUrl = 'https://api.zola.ink/graphql'
```
Comment (Tom-Bonnike) [inline src/login.js:9]: ideally you should constant case such variables : )

File: src/link.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    const json = await fs.readJson(path.join('./', 'package.json'), {
+      throws: false
+    })
+
+    if (
+      !json.zola ||
+      !json.zola.id ||
+      !json.zola.name ||
+      !json.zola.slug ||
+      !json.zola.defaultLocaleId
```
Comment (Tom-Bonnike) [inline src/link.js:17]: mdr

**Review by Tom-Bonnike — APPROVED** [review]

**Pull request merged**